### PR TITLE
feature - added head2head overlay in ready phase

### PR DESCRIPTION
**Pull request by jameshallam93**
* added new overlay to display head to head info for player and opponent, rendered to board container during READY phase
* amended _overlay.scss to include formatting for new component
* applied vs code formatting

### CODE REVIEW

File: packages/app/src/game/module/board/overlays/readyOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { getUserId } from "packages/app/src/auth";
+import { PlayerListPlayer, StreakType } from "packages/models/lib";
+import * as React from "react";
+import { useSelector } from "react-redux";
+import { AppState } from "../../../../store";
+import { BoardOverlay } from "./boardOverlay";
+
+const getStreakType = (player: PlayerListPlayer): string => {
+
+    const streakType = player?.streakAmount === 0 ? ""
```
Comment (Jameskmonger) [inline packages/app/src/game/module/board/overlays/readyOverlay.tsx:10]: Please use an if statement here for readability. Ternaries are great for a single statement but beyond that I find them hard to read (even though you've done a nice job with the line breaks)

```ts
if (!player || player.streakAmount === 0) {
    return "";
}

if (player.streakType === StreakType.WIN) {
    return player.streakAmount === 1 ? "Win" : "Wins";
}

return player.streakAmount === 1 ? "Loss" : "Losses"
```

File: packages/app/src/game/module/board/overlays/readyOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        : player?.streakType === StreakType.WIN ? player?.streakAmount === 1 ? "Win" : "Wins"
+            : player?.streakAmount === 1 ? "Loss" : "Losses"
+
+    return streakType
+}
+
+const getPosition = (player: PlayerListPlayer): number => {
+    const position = useSelector((state: AppState) => {
```
Comment (Jameskmonger) [inline packages/app/src/game/module/board/overlays/readyOverlay.tsx:18]: `useSelector` can only be used in component render functions. To get around this you could make `getPosition` take `(player: PlayerListPlayer, playerList: PlayerListPlayer[])`, and then put the `useSelector` in the component, passing both in here

File: packages/app/src/game/module/board/boardContainer.tsx [hunk 3]
```diff
@@ -97,7 +99,7 @@ const BoardContainer: React.FunctionComponent<{ showNowPlaying?: boolean }> = ({
                         )}
                     />
                 </div>
-
+                {inReadyPhase && <ReadyOverlay/>}
```
Comment (Jameskmonger) [inline packages/app/src/game/module/board/boardContainer.tsx:102]: Put this logic into the overlay itself, it stops the `BoardContainer` from being concerned here when it doesn't need to be

**Review by Jameskmonger — CHANGES_REQUESTED** [review]

File: packages/app/src/game/module/board/overlays/readyOverlay.tsx (hunk 1 above)
Comment (jameshallam93) [reply packages/app/src/game/module/board/overlays/readyOverlay.tsx]: *Changed to an if statement for getStreakType

File: packages/app/src/game/module/board/overlays/readyOverlay.tsx (hunk 2 above)
Comment (jameshallam93) [reply packages/app/src/game/module/board/overlays/readyOverlay.tsx]: * Moved useSelector as suggested and passed playerList to getPosition function

File: packages/app/src/game/module/board/boardContainer.tsx (hunk 3 above)
Comment (jameshallam93) [reply packages/app/src/game/module/board/boardContainer.tsx]: * removed inReadyPhase variable from BoardContainer and repositioned logic in component itself

**Comment by jameshallam93** [comment]
Have realised that I was using an exported function (getUserId) that has since been deprecated - most of the data I was using from PlayerListPlayer can be found in PlayerInfo, but will try and find a way of getting player position that doesn't rely on the PlayerList order

**Pull request merged**